test/lib/test_services.cc

```diff
@@ -136,7 +136,7 @@ table_for_tests::table_for_tests(sstables::sstables_manager& sstables_manager, s
     _data->cfg.cf_stats = &_data->cf_stats;
     _data->cfg.enable_commitlog = false;
     _data->cm.enable();
-    _data->cf = make_lw_shared<replica::column_family>(_data->s, _data->cfg, make_lw_shared<replica::storage_options>(), _data->cm, sstables_manager, _data->cl_stats, _data->tracker, nullptr);
+    _data->cf = make_lw_shared<replica::column_family>(_data->s, _data->cfg, make_lw_shared<replica::storage_options>(), _data->cm, sstables_manager, _data->cl_stats, sstables_manager.get_cache_tracker(), nullptr);
```

Comment:
This is somewhat random. sstables_manager happens to have a cache_tracker, tomorrow it might not.

I don't mind keeping the patch, but there's a difference between inheriting a dependency by accident or because it's really needed. Kind of like inheriting a #include because some dependent header happened to include it.

Reply:
I agree that this place is pretty much a coincidence. But it's inheriting the inverse logic of sstables::test_env creating a table object. I'm fixing this inversion in another branch, so this part should also improve eventually.